### PR TITLE
allow constructing SecretKey from buffer

### PR DESCRIPTION
**Pull request by staltz**
Context: I need to fetch the group subfeed, but the group subfeed has the secret key bytes as the `feedpurpose`. What I have is the groupId, and from that I can fetch (from keyring) the secret key bytes, but then I need to create a `SecretKey` class instance.

Asking for a review from whoever sees this first.

### CODE REVIEW

**Comment by staltz** [comment]
Done

**Comment by staltz** [comment]
I ran tests actually, and they passed. Unfortunately there were no tests for SecretKey, and I didn't add any new tests. But I'm in a hurry.

File: secret-key.js [hunk 1]
```diff
@@ -5,9 +5,13 @@ const na = require('sodium-universal')
...
 
 module.exports = class SecretKey {
-  constructor (length) {
-    this.key = na.sodium_malloc(length || na.crypto_secretbox_KEYBYTES)
-    na.randombytes_buf(this.key)
+  constructor (lengthOrBuffer) {
+    if (Buffer.isBuffer(lengthOrBuffer)) {
+      this.key = lengthOrBuffer
+    } else {
+      this.key = na.sodium_malloc(length || na.crypto_secretbox_KEYBYTES);
```
Comment (staltz) [inline secret-key.js:12]: Aaaah, annoying that I missed this detail: `length` is not defined.

Comment (arj03) [reply secret-key.js:12]: Oh, this repo doesn't have automated tests? :) I saw no errors and assumed this must have passed the tests, or maybe there is no test of this constructor. Anyway glad you fixed it :)

Comment (staltz) [reply secret-key.js:12]: Yeah no CI tests :smiling_face_with_tear:

Comment (mixmix) [reply secret-key.js:12]: SecretKey came from hmmm another module. Likely tested there. Though maybe integration test not unit test

**Pull request merged**